nest-forms-backend/src/forms/forms.service.ts

```diff
@@ -299,35 +254,30 @@ export default class FormsService {
             state: true,
             error: true,
             formDataJson: true,
-            schemaVersionId: true,
-            schemaVersion: {
-              select: {
-                messageSubjectFormat: true,
-                schema: true,
-                uiSchema: true,
-              },
-            },
+            formDefinitionSlug: true,
           },
         })
       : []
 
     const dataWithLatestFlag: GetFormResponseSimpleDto[] = []
     Object.values(data).forEach((form) => {
-      const isLatestSchemaVersionForSlug =
-        form.schemaVersionId === form.schemaVersion.schema.latestVersionId
-      const messageSubject = getSubjectTextFromForm(form)
+      const formDefinition = getFormDefinitionBySlug(form.formDefinitionSlug)
+      if (!formDefinition) {
+        throw this.throwerErrorGuard.NotFoundException(
```

Comment:
We should probably not throw this error. If we delete some form definition (not really resolved now), and the user has in in the list of their forms, this will cause the route to return 404, which will NOT display all the other forms for the user. Probably some non-existing form flag is better solution.

Reply:
If we delete some form definition, should not we delete all of its forms as well? Because after deleting we will miss all of the information about the form definition, including name etc. Because of this I would say that when we want to delete some form definition in the future, we just add some `isDeleted: true` flag to it, which will cause that this will not error, since it will find it by the slug. Then just some logic checking if the isDeleted flag is present must be implemented.

Reply:
Fair enough, this is not an use case we support now.